### ISSUE TITLE
[bug] `no-useless-fragment` false positive when using &nbsp;

### ISSUE THREAD

**Issue by michaelfaith** [Rel1cx/eslint-react#1035]
### Describe the bug

I found one case where the `no-useless-fragment` rule was incorrectly flagging a fragment that included both a single component accompanied by a `&nbsp;`.  After running the autofix, it left it in a state that resulted in parsing errors.

Original code
```ts
return (
  <>
    <SomeComponent />
    &nbsp;
  </>
);

```

"Fixed" code
```ts
return (
  <SomeComponent />
  &nbsp;
);
```

### Expected behavior

Should consider html entities when assessing whether a fragment is needed or not.

### Platform and versions

```Text
Node 22.11
ESLint 9.22.0
```

### Stack trace

```Text

```
